### PR TITLE
New template "r-ver"

### PR DESCRIPTION
**Pull request by eitsupi**
So far I have chosen a format that does not have a Dockerfile to match many of the devcontainers templates and formats.
https://github.com/devcontainers/templates/tree/main/src

I don't know if this really works as there is no way to use this template, but the format is probably correct as it passes the test copied from https://github.com/devcontainers/templates.

### CODE REVIEW

File: src/r/devcontainer-template.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
+	"id": "r",
+	"version": "0.1.0",
+	"name": "R",
+	"description": "Editing R codes on a container. Includes many R packages from the start.",
```
[the remark this answers is not in the document]
Comment (cboettig) [reply src/r/devcontainer-template.json:5]: Nice.  For the description, what do you think about:

> A container-based development environment for the R language. Includes many commonly used R packages.

**Comment by cboettig** [comment]
this is cool.  It would be really nice if there was a good way to test the template.  

I was testing out the setup in https://github.com/espm-157/devcontainers-rstudio/, and definitely starting to enjoy working directly in the vscode environment.  However working with the free-tier 2 GB ram VMs is really constraining without having packages pre-installed (discovered I could not install the dev version of `terra` package from source without crashing due to OOM).  For dev-containers to be effective on the free tier having a convenient way to start from pre-built images with most necessary packages installed might be important.

**Review by cboettig — APPROVED** [review]

**Comment by eddelbuettel** [comment]
(<cough, cough>  r2u.  No compiling or building, all of CRAN. I still haven't set down for the vscode demo/setup but doing with gitpod.io was a breeze.)

**Comment by eitsupi** [comment]
> (<cough, cough> r2u. No compiling or building, all of CRAN. I still haven't set down for the vscode demo/setup but doing with gitpod.io was a breeze.)

On amd64, I believe the images used in this template also does binary installs.
I don't think there is a way to do a binary install of the development version packages (Windows and macOS can install from R-universe......)

As for the pain of installing from CRAN, bspm is definitely easier, but installing packages using pak bypasses bspm, so I had to write a separate script to install the R package, and I have been putting off preparing that for now.
rocker-org/devcontainer-images#5

**Comment by eddelbuettel** [comment]
Sorry but that misses the point: add-ons like `pak` will _never_ get system dependencies automated.  Never.  They can't.

`r2u` does that by building on the breadth offered by RSPM and by then adding the _missing depth_ of using the in-distro tools to add system dependencies.  This goes both ways:  on 'user library addition', system libraries get added (the `stringi -> libicu*` or `*gsl` problems).  Equally importantly, `apt` is now connected to the use by the R packages _and will never uninstall a (versioned) shared library used by an R package_ because "now it knows".

No `pak` or `rig` or ... tool can do that until it integrates with the OS.  Which RSPM likely never will as it focusses on _breadth_. But you build _depth_ here with the devcontainers so you _could_ (if you chose to) take advantage of it.

File: src/r/devcontainer-template.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (eitsupi) [reply src/r/devcontainer-template.json]: Thanks, that looks good!

**Comment by eitsupi** [comment]
> `r2u` does that by building on the breadth offered by RSPM and by then adding the _missing depth_ of using the in-distro tools to add system dependencies. This goes both ways: on 'user library addition', system libraries get added (the `stringi -> libicu*` or `*gsl` problems). Equally importantly, `apt` is now connected to the use by the R packages _and will never uninstall a (versioned) shared library used by an R package_ because "now it knows".
> 
> No `pak` or `rig` or ... tool can do that until it integrates with the OS. Which RSPM likely never will as it focusses on _breadth_. But you build _depth_ here with the devcontainers so you _could_ (if you chose to) take advantage of it.

As I have stated many times I fully understand the benefits of it. I just didn't tackle it first.
Of course I will work on this later.

**Comment by eitsupi** [comment]
@eddelbuettel I have created an issue to track this. rocker-org/devcontainer-features#66
I have seen Japanese users trying to use bspm on Google Colab and really understand the importance of this.

**Comment by eitsupi** [comment]
It seems to make more sense to use `r-ver` as the template name, given the need to separate the documentation when adding a template based on the R installation via apt later.

**Comment by eitsupi** [comment]
Note that, the current VSCode Dev Containers does not reflect upstream's vscode customization settings, so there seems to be no VSCode settings for R on the container launched using this template.
However, looking at https://github.com/devcontainers/templates, the other templates have the same structure, so I would like to leave it as it is first.

**Comment by cboettig** [comment]
Thanks gents, didn't mean to kick off another round of discussion of binary install mechanisms in my comment above, but just for clarity I was specifically referring to the use case of when pre-built binaries are not available upstream.  For example, at this moment I'd very much like the 54 students in my classroom to be able to deploy _the current github version of terra_ (v1.6.31) but none of our cloud environments (the berkeley data hub, the GitHub Codespaces images, RStudio cloud free instances etc) have enough RAM to build from source.  My thought was to just prebuild a docker image and push it to docker hub to use that as the base image in codespaces...  better ideas welcome

**Comment by eitsupi** [comment]
I believe Codespaces has the ability to build an image without pushing it to the repository, but still couldn't build it?
https://docs.github.com/en/codespaces/prebuilding-your-codespaces/about-github-codespaces-prebuilds

**Comment by eddelbuettel** [comment]
> My thought was to just prebuild a docker image

Similarly/simpler (to me) I have sometimes created ad-hoc .deb packages (from open source parts) on Launchpad. Then it is back to the binary...

**Comment by eitsupi** [comment]
@cboettig R-universe now upload binary packages for amd64 jammy, so you can install dev version of `terra` with this command `install.packages("terra", repos = "https://rspatial.r-universe.dev/bin/linux/jammy/4.2")`.

**Comment by cboettig** [comment]
Thanks @eitsupi, that's really handy! 

I also discovered that you were correct about the devcontainer build having more free memory, so I can just drop a one-line Dockerfile into the repo and codespaces can successfully run the install before the instance deploys: https://github.com/espm-157/devcontainers-rstudio/tree/main/.devcontainer.  Obviously this slows the initial deploy down but seems a reasonable compromise here.

**Pull request merged**